


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html id="htmlId">
<head>
  <title>Coverage Report :: ExceptionInInitializerError</title>
  <style type="text/css">
    @import "../../.css/coverage.css";
  </style>
</head>

<body>
<div class="header"></div>

<div class="content">
<div class="breadCrumbs">
    [ <a href="../../index.html">all classes</a> ]
    [ <a href="../index.html">java.lang</a> ]
</div>

<h1>Coverage Summary for Class: ExceptionInInitializerError (java.lang)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">ExceptionInInitializerError</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 5)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 9)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<div class="sourceCode"><i>1</i>&nbsp;/*
<i>2</i>&nbsp; * Copyright (c) 1996, 2000, Oracle and/or its affiliates. All rights reserved.
<i>3</i>&nbsp; * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
<i>4</i>&nbsp; *
<i>5</i>&nbsp; *
<i>6</i>&nbsp; *
<i>7</i>&nbsp; *
<i>8</i>&nbsp; *
<i>9</i>&nbsp; *
<i>10</i>&nbsp; *
<i>11</i>&nbsp; *
<i>12</i>&nbsp; *
<i>13</i>&nbsp; *
<i>14</i>&nbsp; *
<i>15</i>&nbsp; *
<i>16</i>&nbsp; *
<i>17</i>&nbsp; *
<i>18</i>&nbsp; *
<i>19</i>&nbsp; *
<i>20</i>&nbsp; *
<i>21</i>&nbsp; *
<i>22</i>&nbsp; *
<i>23</i>&nbsp; *
<i>24</i>&nbsp; */
<i>25</i>&nbsp;
<i>26</i>&nbsp;package java.lang;
<i>27</i>&nbsp;
<i>28</i>&nbsp;/**
<i>29</i>&nbsp; * Signals that an unexpected exception has occurred in a static initializer.
<i>30</i>&nbsp; * An &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; is thrown to indicate that an
<i>31</i>&nbsp; * exception occurred during evaluation of a static initializer or the
<i>32</i>&nbsp; * initializer for a static variable.
<i>33</i>&nbsp; *
<i>34</i>&nbsp; * &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
<i>35</i>&nbsp; * the general purpose exception-chaining mechanism.  The &quot;saved throwable
<i>36</i>&nbsp; * object&quot; that may be provided at construction time and accessed via
<i>37</i>&nbsp; * the {@link #getException()} method is now known as the &lt;i&gt;cause&lt;/i&gt;,
<i>38</i>&nbsp; * and may be accessed via the {@link Throwable#getCause()} method, as well
<i>39</i>&nbsp; * as the aforementioned &quot;legacy method.&quot;
<i>40</i>&nbsp; *
<i>41</i>&nbsp; * @author  Frank Yellin
<i>42</i>&nbsp; * @since   1.1
<i>43</i>&nbsp; */
<i>44</i>&nbsp;public class ExceptionInInitializerError extends LinkageError {
<i>45</i>&nbsp;    /**
<i>46</i>&nbsp;     * Use serialVersionUID from JDK 1.1.X for interoperability
<i>47</i>&nbsp;     */
<i>48</i>&nbsp;    private static final long serialVersionUID = 1521711792217232256L;
<i>49</i>&nbsp;
<i>50</i>&nbsp;    /**
<i>51</i>&nbsp;     * This field holds the exception if the
<i>52</i>&nbsp;     * ExceptionInInitializerError(Throwable thrown) constructor was
<i>53</i>&nbsp;     * used to instantiate the object
<i>54</i>&nbsp;     *
<i>55</i>&nbsp;     * @serial
<i>56</i>&nbsp;     *
<i>57</i>&nbsp;     */
<i>58</i>&nbsp;    private Throwable exception;
<i>59</i>&nbsp;
<i>60</i>&nbsp;    /**
<i>61</i>&nbsp;     * Constructs an &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; with
<i>62</i>&nbsp;     * &lt;code&gt;null&lt;/code&gt; as its detail message string and with no saved
<i>63</i>&nbsp;     * throwable object.
<i>64</i>&nbsp;     * A detail message is a String that describes this particular exception.
<i>65</i>&nbsp;     */
<b class="nc"><i>66</i>&nbsp;    public ExceptionInInitializerError() {</b>
<b class="nc"><i>67</i>&nbsp;        initCause(null);  // Disallow subsequent initCause</b>
<i>68</i>&nbsp;    }
<i>69</i>&nbsp;
<i>70</i>&nbsp;    /**
<i>71</i>&nbsp;     * Constructs a new &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; class by
<i>72</i>&nbsp;     * saving a reference to the &lt;code&gt;Throwable&lt;/code&gt; object thrown for
<i>73</i>&nbsp;     * later retrieval by the {@link #getException()} method. The detail
<i>74</i>&nbsp;     * message string is set to &lt;code&gt;null&lt;/code&gt;.
<i>75</i>&nbsp;     *
<i>76</i>&nbsp;     * @param thrown The exception thrown
<i>77</i>&nbsp;     */
<b class="nc"><i>78</i>&nbsp;    public ExceptionInInitializerError(Throwable thrown) {</b>
<b class="nc"><i>79</i>&nbsp;        initCause(null);  // Disallow subsequent initCause</b>
<b class="nc"><i>80</i>&nbsp;        this.exception = thrown;</b>
<i>81</i>&nbsp;    }
<i>82</i>&nbsp;
<i>83</i>&nbsp;    /**
<i>84</i>&nbsp;     * Constructs an ExceptionInInitializerError with the specified detail
<i>85</i>&nbsp;     * message string.  A detail message is a String that describes this
<i>86</i>&nbsp;     * particular exception. The detail message string is saved for later
<i>87</i>&nbsp;     * retrieval by the {@link Throwable#getMessage()} method. There is no
<i>88</i>&nbsp;     * saved throwable object.
<i>89</i>&nbsp;     *
<i>90</i>&nbsp;     *
<i>91</i>&nbsp;     * @param s the detail message
<i>92</i>&nbsp;     */
<i>93</i>&nbsp;    public ExceptionInInitializerError(String s) {
<b class="nc"><i>94</i>&nbsp;        super(s);</b>
<b class="nc"><i>95</i>&nbsp;        initCause(null);  // Disallow subsequent initCause</b>
<i>96</i>&nbsp;    }
<i>97</i>&nbsp;
<i>98</i>&nbsp;    /**
<i>99</i>&nbsp;     * Returns the exception that occurred during a static initialization that
<i>100</i>&nbsp;     * caused this error to be created.
<i>101</i>&nbsp;     *
<i>102</i>&nbsp;     * &lt;p&gt;This method predates the general-purpose exception chaining facility.
<i>103</i>&nbsp;     * The {@link Throwable#getCause()} method is now the preferred means of
<i>104</i>&nbsp;     * obtaining this information.
<i>105</i>&nbsp;     *
<i>106</i>&nbsp;     * @return the saved throwable object of this
<i>107</i>&nbsp;     *         &lt;code&gt;ExceptionInInitializerError&lt;/code&gt;, or &lt;code&gt;null&lt;/code&gt;
<i>108</i>&nbsp;     *         if this &lt;code&gt;ExceptionInInitializerError&lt;/code&gt; has no saved
<i>109</i>&nbsp;     *         throwable object.
<i>110</i>&nbsp;     */
<i>111</i>&nbsp;    public Throwable getException() {
<b class="nc"><i>112</i>&nbsp;        return exception;</b>
<i>113</i>&nbsp;    }
<i>114</i>&nbsp;
<i>115</i>&nbsp;    /**
<i>116</i>&nbsp;     * Returns the cause of this error (the exception that occurred
<i>117</i>&nbsp;     * during a static initialization that caused this error to be created).
<i>118</i>&nbsp;     *
<i>119</i>&nbsp;     * @return  the cause of this error or &lt;code&gt;null&lt;/code&gt; if the
<i>120</i>&nbsp;     *          cause is nonexistent or unknown.
<i>121</i>&nbsp;     * @since   1.4
<i>122</i>&nbsp;     */
<i>123</i>&nbsp;    public Throwable getCause() {
<b class="nc"><i>124</i>&nbsp;        return exception;</b>
<i>125</i>&nbsp;    }
<i>126</i>&nbsp;}
</div>
</div>

<div class="footer">
    
    <div style="float:right;">generated on 2018-11-19 23:13</div>
</div>
</body>
</html>
